src/components/Slider/SliderBase/SliderThumb.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import { DialogPositions } from "../../../constants";
+import { NOOP } from "../../../utils/function-utils";
+import Tooltip from "../../Tooltip/Tooltip";
+import { bem } from "../SliderHelpers";
+import { useSliderActions, useSliderSelection, useSliderUi } from "../SliderContext";
+
+const tooltipShowDelay = 300;
+const tooltipPosition = DialogPositions.TOP;
+
+const SliderThumb = ({ className, index, onMove, position }) => {
+  const { max, min, ranged, value: valueOrValues, valueText: valueOrValuesText } = useSliderSelection();
+  const value = ranged ? valueOrValues[index] : valueOrValues;
+  const valueText = ranged ? valueOrValuesText[index] : valueOrValuesText;
+  const { active, ariaLabel, ariaLabelledby, disabled, dragging, focused, shapeTestId, showValue } = useSliderUi();
+  const { setActive, setFocused, setDragging } = useSliderActions();
+  const ref = useRef(null);
+
+  function handleBlur() {
```

Comment:
I think that according to our new conventions from the meeting yesterday the functions should be defined outside the component. Am I right?

Reply:
No, it not pure function it uses components functions: setFocused and setActive